core/Pos/Core/Context.hs

```diff
@@ -9,9 +24,57 @@ module Pos.Core.Context
 
 import           Universum
 
-import           Pos.Core.Address (addressHash, makePubKeyAddress)
-import           Pos.Core.Types   (Address, StakeholderId)
-import           Pos.Crypto       (PublicKey, SecretKey, toPublic)
+import           Control.Lens       (makeLenses)
+import           Data.Reflection    (Given (..), give)
+
+import           Pos.Core.Address   (addressHash, makePubKeyAddress)
+import           Pos.Core.Types     (Address, BlockCount (..), StakeholderId, SlotCount)
+import           Pos.Crypto         (PublicKey, SecretKey, toPublic)
+import           Pos.Core.Constants (staticBlkSecurityParam)
+
+-- | Core constants. They should be really constant and never change.
+data CoreConstants = CoreConstants
+    { _ccBlkSecurityParam :: !BlockCount
+    }
+
+makeLenses ''CoreConstants
+
+type HasCoreConstants = Given CoreConstants
+
+giveConsts :: BlockCount -> (HasCoreConstants => r) -> r
```

Comment:
We already have a data type. Functions with 100500 arguments are ugly and sooner or later this function will start having 100500 arguments and someone will have to refactor it. It's better to do it now than to postpone it. Also there is no advantage in your approach.